### ISSUE TITLE
TestTaskRunRetry is flaky

### ISSUE THREAD

**Issue by jlpettersson** [tektoncd/pipeline#3021]
TestTaskRunRetry is flaky.

I run into this several times for a PR today and for another PR yesterday.

examples: 

* https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/2885/pull-tekton-pipeline-integration-tests/1288049842701996034/
* https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/2885/pull-tekton-pipeline-integration-tests/1288056511737630720/

/kind flake

**Comment by thomaschandler** [comment]
@jlpettersson I wonder if this might be related to the [garbage collection of failed pods](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-garbage-collection). I'm unsure if GKE has a different garbage collection process than a self hosted k8s setup.

I'm thinking of updating `TestTaskRunRetry` to wait for pod state changes, at which point it will record the pod information for the failed pod, which can later be asserted once the `PipelineRun` is complete. Think this is worth a shot?

**Comment by ImJasonH** [comment]
This is happening again/still: https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_pipeline/3494/pull-tekton-pipeline-integration-tests/1324621677065146370/

```

TestTaskRunRetry: retry_test.go:119: BUG: Found 5 Pods, want 6

```

**Comment by chmouel** [comment]
On our downstream nightly CI we had quite a lot of issues with those as well, and could not figure out (at that time) why our CI which is openshift based  consuming more memory  and was affected by it and not on upstream, so we had to disable it : 

https://github.com/openshift/tektoncd-pipeline/blob/master/openshift/patches/0001-Remove-task-retry-since-too-flaky.patch

It was never happening when running it manually (the testsuit running on th same cluster where it was failing)  so that was hard to debug,

**Comment by bobcatfish** [comment]
Sounds like this has been happening _less_ but still happening so:

/remove-lifecycle rotten
